postProcessing/Folders/CosmicIntegration/PythonScripts/totalMassEvolvedPerZ.py

```diff
@@ -159,26 +159,27 @@ def retrieveMassEvolvedPerZ(path, fileName):
 
 
 
-def totalMassEvolvedPerZ(path=None, fileName=None, Mlower=None, Mupper=None, binaryFraction=0.7, \
+def totalMassEvolvedPerZ(path=None, fileName=None, M1lower=None, M1upper=None, M2lower=None, binaryFraction=0.7, \
                          x1=0.01, x2=0.08, x3=0.5, x4=200., a1=-0.3, a2=-1.3, a3=-2.3, C1=1., Mmax=200):
```

Comment:
Is it clear to anyone else what the difference between `M1upper` and `Mmax` is? In the same vein, we don't seem to pass `M1lower` to `createSampleUniverse` (leaving `M1min=0.01` as default)

Reply:
Is that a rhetorical question or a real one?  From the code, Mmax is the maximum stellar mass in the Universe (the upper cutoff of the Kroupa IMF), while Mupper or M1upper is the maximum mass simulated in a COMPAS run.  But I think you should feel free to use more descriptive names, @TomWagg , as long you keep all the functions, classes, and notebooks mutually consistent.

Reply:
Ah my apologies, not a rhetorical question but perhaps a misguided one sorry 😬 

I got confused because `x4` also seems to be the upper cutoff the Kroupa IMF but I guess the `x4` just says when Kroupa stops and `Mmax` is used to tweak this in some way? I honestly still don't quite understand what the difference between the `x4` and `Mmax` is (which is why I wondered if `Mmax` is related to `M1upper`)

Sorry for my ignorance!